tools/rna_tools/cleaverna/cleaverna.xml

```diff
@@ -0,0 +1,636 @@
+            <!--
+            <param name="include_config" type="boolean" checked="false" label="Parameters configuration file (parameters.cfg)" help="Keep the configuration file used for this run."/>
+            <param name="include_metrics" type="boolean" checked="false" label="ML performance metrics (ML_metrics.csv)" help="Keep the machine learning metrics file from prediction mode, if produced."/>
+            <param name="include_training_data" type="boolean" checked="false" label="ML training data (ML_train.csv)" help="Keep the full training data matrix from prediction mode, if produced."/>
+            <param name="include_classification" type="boolean" checked="false" label="Balanced classification results (balanced_classification.csv)" help="Keep the balanced classification results from prediction mode, if produced."/>
+            -->
+    
+    <outputs>
+        <!-- Training mode: only user_merged_features and parameters_config are produced -->
+        <data name="user_merged_features" format="csv" from_work_dir="outputs/*_user_merged_num.csv" label="Training features">
+            <filter>mode_selection['mode'] == 'training'</filter>
+        </data>
+        <data name="parameters_config_train" format="txt" from_work_dir="outputs/parameters.cfg" label="Parameters configuration file (train)">
+            <filter>mode_selection['mode'] == 'training'</filter>
+        </data>
+        <!-- Prediction mode: all other outputs -->
+        <data name="ml_metrics" format="csv" from_work_dir="outputs/*_ML_metrics.csv" label="ML Metrics">
+            <filter>mode_selection['mode'] == 'prediction'</filter>
+        </data>
+        <data name="ml_train" format="csv" from_work_dir="outputs/*_ML_train.csv" label="ML Train">
+            <filter>mode_selection['mode'] == 'prediction'</filter>
+        </data>
+        <data name="ml_train_feature_set" format="csv" from_work_dir="outputs/*_ML_train_feature_set.csv" label="ML Train Feature Set">
+            <filter>mode_selection['mode'] == 'prediction'</filter>
+        </data>
+        <data name="balanced_classification" format="csv" from_work_dir="outputs/*_balanced_classification.csv" label="Balanced Classification">
+            <filter>mode_selection['mode'] == 'prediction'</filter>
+        </data>
+        <data name="cleavage_predictions" format="csv" from_work_dir="outputs/*_CleaveRNA_output.csv" label="Cleavage predictions">
+            <filter>mode_selection['mode'] == 'prediction'</filter>
+        </data>
+        <data name="parameters_config_pred" format="txt" from_work_dir="outputs/parameters.cfg" label="Parameters configuration file (predict)">
+            <filter>mode_selection['mode'] == 'prediction'</filter>
+        </data>
+    </outputs>
+    <tests>
+        <!-- Updated test: Training mode with all required fields -->
+        <test expect_num_outputs="2">
```

Comment:
we need many more tests here, for each conditional at least one ... again the galaxy-lanuage server can help here and generate tests.